test/parallel/test-dgram-implicit-bind-failure.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const dgram = require('dgram');
+const dns = require('dns');
+
+// Monkey patch dns.lookup() so that it always fails.
+dns.lookup = function(address, family, callback) {
+  process.nextTick(() => { callback(new Error('fake DNS')); });
+};
+
+const socket = dgram.createSocket('udp4');
+let dnsFailures = 0;
+let sendFailures = 0;
+
+process.on('exit', () => {
+  assert.strictEqual(dnsFailures, 3);
+  assert.strictEqual(sendFailures, 3);
+});
+
+socket.on('error', (err) => {
```

Comment:
No need in this case, because two types of errors are counted and checked on process exit.